Wordle.Api/Wordle.Api/Services/PlayerService.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Services;
+
+public class PlayerService
+{
+    public WordleDbContext Db { get; set; }
+    
+    public PlayerService(WordleDbContext db)
+    {
+        Db = db;
+    }
+    
+    public async Task<Player[]> GetTopPlayers(int numberOfPlayers)
+    {
+        return await Db.Players.OrderByDescending(p => p.AverageAttempts).Take(numberOfPlayers).ToArrayAsync();
+    }
+    
+    public async Task AddPlayer(Player player)
+    {
+        if (!Db.Players.Contains(player))
+        {
+            await Db.Players.AddAsync(player);
+            await Db.SaveChangesAsync();
+        }
+        
+    }
```

Comment:
I think locks are important here. It's possible two users enter the same name and finish a game at the same time. Race condition is possible here.

Reply:
You are right, locks are necessary here. Again, it is just an unfinished assignment.